embasi/embedding.py

```diff
@@ -906,3 +919,173 @@ def run(self):
+
+        start = time.time()
+
+        ediff = 1.0
+        etot_prev = 0.0
+        max_cycle = 50
+        n_cycle = 0
+
+        cwd = os.path.join(os.getcwd(),self.run_dir[2:])
+    
+        MU0_opot  = os.path.join(cwd, "MU0/output_potential_00000000")
+        F2A1_opot = os.path.join(cwd, "F2A1/output_potential_00000000")
+        F1A2_opot = os.path.join(cwd, "F1A2/output_potential_00000000")
+
+        F2A1_npot = os.path.join(cwd, "F2A1/output_potential_00000000_new")
+        F1A2_npot = os.path.join(cwd, "F1A2/output_potential_00000000_new")
+
+        MU0_oden  = os.path.join(cwd, "MU0/output_density_00000000")
+        F2A1_oden = os.path.join(cwd, "F2A1/output_density_00000000")
+        F1A2_oden = os.path.join(cwd, "F1A2/output_density_00000000")
+
+        F2A1_nden = os.path.join(cwd, "F2A1/output_density_00000000_new")
+        F1A2_nden = os.path.join(cwd, "F1A2/output_density_00000000_new")
+
+        # Outer SCF loop
+        fdet_file = open("fdet.txt", "w")
+        while np.abs(ediff) > 1.e-6 and n_cycle < max_cycle:
+            n_cycle += 1
+            print("SCF cycle: ", n_cycle)
+
+            if n_cycle == 1:
+                self.MU0.run()
+                try:
+                    tmp = os.path.join(self.run_dir, "F2A1")
+                    os.mkdir(tmp)
+                except FileExistsError:
+                    root_print("Directory F2A1 already exists")
+                try:
+                    tmp = os.path.join(self.run_dir, "F1A2")
+                    os.mkdir(tmp)
+                except FileExistsError:
+                    root_print("Directory F1A2 already exists")
+
+                shutil.copy(MU0_oden, F1A2_oden)
+                shutil.copy(MU0_opot, F1A2_opot)
+
+            # Cluster Calculation
+            self.F1A2.run()
+            shutil.move(F1A2_nden, F2A1_oden)
+            shutil.move(F1A2_npot, F2A1_opot)
+
+            # Environment Calculation
+            self.F2A1.run()
+            shutil.move(F2A1_nden,F1A2_oden)
+            shutil.move(F2A1_npot, F1A2_opot)
+
+            if n_cycle == max_cycle:
+                root_print("Max SCF cycles reached")
+                break
+
+            etot_current = self.F2A1.total_energy + self.F1A2.total_energy
+            fdet_file.write("__________________________\n")
+            fdet_file.write(f"FDET Cycle: {n_cycle}\n")
+            fdet_file.write(f"etot_F2A1 {self.F2A1.total_energy}\n")
+            fdet_file.write(f"etot_F1A2 {self.F1A2.total_energy}\n")
+            fdet_file.write(f"etot_current {etot_current}\n")
+            ediff = etot_current - etot_prev
+            etot_prev = etot_current
+            fdet_file.write(f"ediff {ediff}\n")
+            fdet_file.write("__________________________\n")
+            fdet_file.flush()
+            #print(ediff)
+
+
+        end = time.time()
+        fdet_file.close()
+
+        root_print(f"Total time (s): {end-start}")
+
```

Comment:
I think a small final output section could be useful here - I am also now thinking we need some unified output syntax. Either way, just a few output lines here to let the user know the final energy they need to care about and the fragment energies.